tests/functional/api/test_groups.py

```diff
@@ -308,3 +308,12 @@ def test_group_saml_group_links(group):
     group.saml_group_links.create(
         {"saml_group_name": "saml-group-1", "access_level": 10}
     )
+
+
+@pytest.mark.gitlab_premium
+def test_group_service_account(group):
+    hook = group.service_accounts.create(
+        {"name": "gitlab-service-account", "username": "gitlab-service-account"}
+    )
+    assert hook.name == "gitlab-service-account"
+    assert hook.username == "gitlab-service-account"
```

Comment:
Not sure if this was intentional but this seems like it would be a better fit:

```suggestion
    account = group.service_accounts.create(
        {"name": "gitlab-service-account", "username": "gitlab-service-account"}
    )
    assert account.name == "gitlab-service-account"
    assert account.username == "gitlab-service-account"
```

Reply:
Not intended. Copy pasta.